File: Laravel/provePHP/books/app/Http/Controllers/BookController.php
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;
use App\Models\Book;
class BookController extends Controller
{
    public function list(){
        $books = Book::all();
        return view('list', compact('books'));
    }

    public function aggiungi(Request $request){
        $book = $request->all();
        Book::create($book);
        return redirect('/books');
    }

    public function detail($id){
        $book = Book::find($id);
        //oppure ->get()  
        //oppure Book::where('id',$id)->first();
        return view('detail', compact('book'));
    }

    public function update(Request $request, Book $book){
        $book->update($request->all());   
        return redirect()->route('book.detail', $book->id);
    }

    public function delete(Book $book){
        $book->delete();
        return redirect()->route('book.list');
    }
}

File: SpringBoot/proveEsame/authors_books/src/main/java/com/example/authors_books/controllers/BookController.java
package com.example.authors_books.controllers;

import java.util.List;

import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.RequestMapping;

import com.example.authors_books.data.AuthorRepository;
import com.example.authors_books.data.BookRepository;
import com.example.authors_books.models.Author;
import com.example.authors_books.models.Book;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;


@RequestMapping("/books")
@Controller
public class BookController {

    private AuthorRepository authorRepository;
    private BookRepository bookRepository;

    public BookController(AuthorRepository authorRepository, BookRepository bookRepository){
        this.authorRepository = authorRepository;
        this.bookRepository = bookRepository;
    }

    @GetMapping
    public String index(Model model) {
        List<Book> books = bookRepository.findAll();
        model.addAttribute("books", books);
        List<Author> authors = authorRepository.findAll();
        model.addAttribute("authors", authors);
        return "book/books";
    }

    @GetMapping("/create/{authorId}")
    public String help_create(@PathVariable Long authorId, Model model) {
        List<Book> books = bookRepository.findAll();
        model.addAttribute("books", books);
        List<Author> authors = authorRepository.findAll();
        model.addAttribute("authors", authors);
        model.addAttribute("authorId", authorId);
        
        return "book/createBook";
    }

    @PostMapping
    public String store(Book book) {
        bookRepository.save(book);
        return "redirect:/books";
    }

    @GetMapping("/{id}")
    public String show(@PathVariable Long id, Model model) {
        Book book = bookRepository.findById(id).orElse(null);
        model.addAttribute("book", book);
        List<Author> authors = authorRepository.findAll();
        model.addAttribute("authors", authors);
        return "book/book";
    }

    @PostMapping("/{id}")
    public String update(Book book) {
        bookRepository.save(book);
        return "redirect:/books/"+book.getId();
    }

    @PostMapping("/delete/{id}")
    public String delete(@PathVariable Long id) {
        Book book = bookRepository.findById(id).orElse(null);
        bookRepository.delete(book);
        return "redirect:/books";
    }
    
}
